github/Repository.py

```diff
@@ -1264,8 +1264,8 @@ def create_git_release(
         :rtype: :class:`github.GitRelease.GitRelease`
         """
         assert isinstance(tag, str), tag
-        assert isinstance(name, str), name
-        assert isinstance(message, str), message
+        assert is_optional(name, str), name
```

Comment:
Should we consider the value of `generate_release_notes`? As you said, it is optional only if `generate_release_notes` is true.